### ISSUE TITLE
Find references misses cross-file usages of instance attributes declared in __init__

### ISSUE THREAD

**Issue by vojtech-homola** [astral-sh/ty#4545]
### Summary

In Zed, “Find all references” works with basedpyright but misses usages in other files when using ty for instance attributes declared inside `__init__`.

Minimal example:

`states.py`:

```python
class State:
    def __init__(self) -> None:
        self.fov: float = 20.0
```

`use.py`:

```python
from states import State

state = State()
print(state.fov)
```

**Expected:** Searching for references to `fov` from either file finds both the declaration and the usage.

**Actual:** Searching from the declaration only finds the declaration. Searching from `use.py` only finds the usage there.

Hover and “Go to definition” work correctly.

The problem also reproduces independently of Zed: direct `ty server` tests using `textDocument/references` with `includeDeclaration: true` succeeded in 0.0.34 but failed in 0.0.35 and 0.0.73. The reproduction used no project configuration; the client set `diagnosticMode: "off"`.

The suspected cause is [astral-sh/ruff#25033](https://github.com/astral-sh/ruff/pull/25033), which introduced a scope-based check that skips cross-file searches for function-scoped definitions. This appears to also exclude instance attributes declared inside `__init__`. The attribution is based on release testing and source inspection; individual commits were not built and tested.

Adding a class-level annotation restores cross-file references in the tested 0.0.73 example:

```python
class State:
    fov: float

    def __init__(self) -> None:
        self.fov = 20.0
```

### Version

ty 0.0.81
